contracts/extensions/DeltaNeutralBasisTradingStrategyExtension.sol

```diff
@@ -0,0 +1,1549 @@
+        );
+
+        invokeManager(address(strategy.basisTradingModule), withdrawCalldata);
+    }
+
+    /**
+     * Calculates chunk rebalance notional and calls `_executeEngageTrades` to open required positions. Used in the rebalance() and iterateRebalance() functions
+     *
+     * return uint256           Calculated notional to trade
+     * return uint256           Total notional to rebalance over TWAP
+     */
+    function _handleRebalance(LeverageInfo memory _leverageInfo, int256 _newLeverageRatio) internal returns(int256, int256) {
+        (
+            int256 chunkRebalanceNotional,
+            int256 totalRebalanceNotional
+        ) = _calculateChunkRebalanceNotional(_leverageInfo, _newLeverageRatio);
+
+        _executeRebalanceTrades(_leverageInfo, chunkRebalanceNotional);
+
+        return (chunkRebalanceNotional, totalRebalanceNotional);
+    }
+
+    /**
+     * Calculate base rebalance units and opposite bound units. Invoke trade on TradeModule to acquire spot asset. Deposit rest of the collateral token to PerpV2
+     * and invoke trade on PerpV2BasisTradingModule to open short perp position.
+     */
+    function _executeEngageTrades(
+        LeverageInfo memory _leverageInfo,
+        int256 _chunkRebalanceNotional
+    )
+        internal
+    {
+        int256 baseRebalanceUnits = _chunkRebalanceNotional.preciseDiv(_leverageInfo.action.setTotalSupply.toInt256());
+
+        uint256 oppositeBoundUnits = _calculateOppositeBoundUnits(
+            baseRebalanceUnits.neg(),
+            _leverageInfo.action,
+            _leverageInfo.slippageTolerance
+        ).fromPreciseUnitToDecimals(collateralDecimals);
+
+        _executeDexTrade(baseRebalanceUnits.abs(), oppositeBoundUnits, true);
+
+        _depositAll();
+
+        _executePerpTrade(baseRebalanceUnits, _leverageInfo);
+    }
+
+    /**
+     * Calculate base rebalance units and opposite bound units. Invoke trade on PerpV2BasisTradingModule to lever/delever perp short position. If levering, withdraw
+     * collateral token and invoke trade on TradeModule to acquire more spot assets. If delevering, invoke trade on TradeModule to sell spot assets and deposit the
+     * recieved collateral token to PerpV2.
+     */
+    function _executeRebalanceTrades(
+        LeverageInfo memory _leverageInfo,
+        int256 _chunkRebalanceNotional
+    )
+        internal
+    {
+        int256 baseRebalanceUnits = _chunkRebalanceNotional.preciseDiv(_leverageInfo.action.setTotalSupply.toInt256());
+        uint256 oppositeBoundUnits = _calculateOppositeBoundUnits(
+            baseRebalanceUnits.neg(),
+            _leverageInfo.action,
+            _leverageInfo.slippageTolerance
+        ).fromPreciseUnitToDecimals(collateralDecimals);
+
+        _executePerpTrade(baseRebalanceUnits, _leverageInfo);
+
+        if (baseRebalanceUnits < 0) {
+            _withdraw(oppositeBoundUnits);
+
+            _executeDexTrade(baseRebalanceUnits.abs(), oppositeBoundUnits, true);
```

Comment:
Noting that there is likely to be some leftover USDC units here since there's a good chance we don't use up all of the `collateralToken`

Reply:
Should we withdraw `oppostiteBoundUnits - currentUnits`? Cause otherwise the left out usdc units keep accumulating during continuous lever ups. However, they are deposited back if we either delever or reinvest.

Reply:
Update: Added `_depositAll` at end of rebalance (levering up branch) in a0d9cd2